aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIOProperties.java

```diff
@@ -439,6 +449,7 @@ public void testApplyS3ServiceConfigurations() {
         ArgumentCaptor.forClass(S3Configuration.class);
 
     Mockito.doReturn(mockA).when(mockA).dualstackEnabled(Mockito.anyBoolean());
+    Mockito.doReturn(mockA).when(mockA).crossRegionAccessEnabled(Mockito.anyBoolean());
```

Comment:
please also add `Assertions.assertThat(s3Configuration.crossRegionAccessEnabled()).isTrue()` at the bottom of this test method

Reply:
@nastra 
`crossRegionAccessEnabled` is an `S3Client` attribute rather than `S3Configuration` (same as `dualstackEnabled`). 
Similar to `dualstackEnabled` attribute (not asserted in this test method), `s3Configuration.crossRegionAccessEnabled()` does not exist.

Reply:
Yeah unfortunately I think getting the value of the attributes from the client isn't possible without doing some reflection hacks. 

What I think we can do is add S3 integration tests though to verify the actual cross region access behavior. That seems like a more useful test to verify that when the property is set we can perform operations across regions as expected. cc @jackye1995 @geruh @rahil-c

Reply:
+1 for integ test, there is already a AWS_CROSS_REGION config there that can be used